src/main/java/org/opensearch/ad/ml/CheckpointDao.java

```diff
@@ -436,44 +441,48 @@ public void deleteModelCheckpointByDetectorId(String detectorID) {
+            .allowNoIndices(true)
+            .ignoreUnavailable(true)
+            .expandWildcards(ExpandWildcard.Open)
+            .conflicts(Conflicts.Proceed)// when current delete happens, previous might not finish.
+                                         // Retry in this case
+            .requestsPerSecond(500L)// throttle delete request
+            .build();
         logger.info("Delete checkpoints of detector {}", detectorID);
-        client.execute(DeleteByQueryAction.INSTANCE, deleteRequest, ActionListener.wrap(response -> {
-            if (response.isTimedOut() || !response.getBulkFailures().isEmpty() || !response.getSearchFailures().isEmpty()) {
+        try {
+            CompletableFuture<DeleteByQueryResponse> deleteResponse = sdkJavaAsyncClient.deleteByQuery(deleteRequest);
+            DeleteByQueryResponse response = deleteResponse.orTimeout(10L, TimeUnit.SECONDS).get();
+            if (response.timedOut() || !response.failures().isEmpty()) {
                 logFailure(response, detectorID);
             }
             // can return 0 docs get deleted because:
             // 1) we cannot find matching docs
             // 2) bad stats from OpenSearch. In this case, docs are deleted, but
             // OpenSearch says deleted is 0.
-            logger.info("{} " + DOC_GOT_DELETED_LOG_MSG, response.getDeleted());
-        }, exception -> {
+            logger.info("{} " + DOC_GOT_DELETED_LOG_MSG, response.deleted());
+
+        } catch (Exception exception) {
             if (exception instanceof IndexNotFoundException) {
                 logger.info(INDEX_DELETED_LOG_MSG + " {}", detectorID);
             } else {
                 // Gonna eventually delete in daily cron.
                 logger.error(NOT_ABLE_TO_DELETE_LOG_MSG, exception);
             }
-        }));
+        }
     }
 
-    private void logFailure(BulkByScrollResponse response, String detectorID) {
-        if (response.isTimedOut()) {
+    private void logFailure(DeleteByQueryResponse response, String detectorID) {
+        if (response.timedOut()) {
             logger.warn(TIMEOUT_LOG_MSG + " {}", detectorID);
-        } else if (!response.getBulkFailures().isEmpty()) {
-            logger.warn(BULK_FAILURE_LOG_MSG + " {}", detectorID);
-            for (BulkItemResponse.Failure bulkFailure : response.getBulkFailures()) {
-                logger.warn(bulkFailure);
-            }
         } else {
-            logger.warn(SEARCH_FAILURE_LOG_MSG + " {}", detectorID);
-            for (ScrollableHitSource.SearchFailure searchFailure : response.getSearchFailures()) {
-                logger.warn(searchFailure);
+            if (!response.failures().isEmpty()) {
+                logger.warn(BULK_FAILURE_LOG_MSG + " {}", detectorID);
+                for (BulkIndexByScrollFailure bulkFailure : response.failures()) {
+                    logger.warn(bulkFailure);
+                }
```

Comment:
This is a little confusing to me. We have `BulkIndexByScrollFailure` but old code has `ScrollableHitSource` which is use for the scrollable results. @saratvemulapalli need some inputs here to verify the changes in this file

Reply:
@VachaShah just wanted to get your input on this. Please correct me if I am wrong, but as we discussed before, the `DeleteByQueryResponse` failures for opensearch-java's `OpenSearchAsyncClient` must be read from the list of `BulkIndexByScrollFailure` and there are no corresponding types for the HighLevelRestClient's `SearchFailure` and `BulkFailure`.

Reply:
Yes for `DeleteByQueryResponse` in the java-client, the failures need to be read from the list of `BulkByIndexScrollFailures`.